openapi/components/schemas/Transactions/TransactionRequest.yaml

```
@@ -1,8 +1,12 @@
 allOf:
-  - $ref: ../Common/CommonTransactionRequest.yaml
   - required:
     - type
   - properties:
+      upsertCustomer:
+        type: boolean
+        default: false
+        writeOnly: true
+        description: If true, it will upsert (create or update) a customer. If false, the `customerId` must already exist.
```

nitpick: I think it would good to be more explicit here

"It" is ambiguous and is open to misinterpretation. I think it would also help to state which variable returns or provides the "true" or "false" value.

Also, I think it would be good to state that if the value is `false`, the related customer is not updated

```suggestion
        description: If the response is `true`, this operation upserts (creates or updates) a customer. If the response is `false`, the `customerId` already exists, and the related customer is not updated.
```